lif/lif.go

```diff
@@ -0,0 +1,93 @@
+package lif
+
+import (
+	"encoding/json"
+	"os"
+)
+
+var resources = make(map[string]any)
+
+type AwsEC2 struct {
+	name    string
+	ports   []string
+	envVars map[string]string
+}
+
+type AwsEC2WithPorts struct {
+	instance *AwsEC2
+}
+
+type AwsEC2WithEnvVars struct {
+	instance *AwsEC2
+}
+
+type AwsEC2Exposes struct {
+}
+
+func NewAwsEC2(name string) *AwsEC2 {
```

Comment:
Just realized the pattern we have here doesn't work. It forces us to call the next function in the chain and that should be optional. I'll fix it later. Just fyi.